mcgj/mcgj.py

```diff
@@ -28,7 +28,22 @@ def index():
 @bp.route("/profile")
 @login_required
 def profile():
-    return render_template("edit_profile.html", user=current_user)
+    user_tracks_query = "SELECT * FROM tracks WHERE user_id = ? AND cue_date IS NOT NULL ORDER BY cue_date DESC LIMIT 50"
+    user_tracks = db.query(sql=user_tracks_query, args=[current_user.id])
+    user_tracks = [Track(row) for row in user_tracks]
+
+    for track in user_tracks:
+        d = track.cue_date.date()
+        track.cue_date = d
+
+    return render_template("edit_profile.html", user=current_user, tracks=user_tracks)
+
+@bp.route("/latest_db", methods=['GET'])
+@login_required
+def latest_db():
+    path = current_app.config['DATABASE']
+    return send_file(path, as_attachment=True, attachment_filename='mcgj-latest.db')
```

Comment:
Any idea how send_file behaves when called on a file actively being written to? (Also researching)

Reply:
send_file won't guarantee transactional semantics, and sqlite files will be in a weird state if they're copied during an open db transaction, so you'll definitely get a corrupt DB! :) There's an online backup API in sqlite that involves creating a new file from a known state, but that also involves locking the whole DB while that's happening.

However, nothing bad will happen to the server's DB, so *shrug*?

Reply:
How about a weekly backup (on a cron), and offering that for download? That way folks won't run into the frustration of randomly corrupt DB for the tradeoff of missing the latest session's data.

Reply:
We can do that?? I love it! But lemme look more into the online backup....

Reply:
Yep, looks like I can get my little raccoon paws into crontab on pythonanywhere 🦝